### ISSUE TITLE
Investigate CampusGroups

### ISSUE THREAD

**Issue by jwflory** [RITlug/tasks#21]
Figure out what we need to do to meet the minimum requirements for Club Office, or if there's systems we should use to help make our club activities more transparent to club office

**Comment by justwheel** [comment]
I'm honestly not sure what we want to do with this one. We meet minimum requirements. There's not much I'm motivated to do in research because I don't know how valuable of an activity this is. I'll lower the priority for this one, but if we don't have anything new by the end of the semester, I will close it as `wontfix`.

**Comment by ct-martin** [comment]
At minimum we should have something more obvious pointing to our website
and such.

--
Christian Martin (Phone)

**Comment by Tjzabel** [comment]
@jwflory @ct-martin I am not sure there is anything else we need to do with this issue. Is there anything you see that is actionable?

**Comment by justwheel** [comment]
I forgot to follow up on this one. Closing as `wontfix` given the previous comment.

**Comment by ct-martin** [comment]
Re-opening. I want to see if we can integrate CampusGroups mailing list with the Google Groups one and also see if I can easily get sign-in done on CampusGroups.

For the mailing list it looks like we can have CampusGroups cc the Google Group, or we might be able to get it to accept emails from the Google Group automatically. I'm making this a priority since RIT has been getting very heavy-handed on marking any email coming from off-campus as potential spam/phishing (even if very clearly legitimate; reputable, big-name sender).

**Comment by ct-martin** [comment]
Based on test from today, it seems that we need to add mailer@campusgroups.com to the Google Group so it can send

@jwflory renamed issue again to include the sign-in part

**Comment by justwheel** [comment]
The address was subscribed, but I don't think we made progress on this yet.

@ct-martin Perhaps it would help to split these issues into separate tasks since they may be completed at different times from each other.

**Comment by ct-martin** [comment]
Current status: @jwflory let me into the RITlug root Gmail temporarily and I set up most of eboard (emails I could remember off-hand) with manager access to the Google Group. Unfortunately, the permission to post can not be added to an individual role, so I need to access the email settings next time I can ~kidnap~ sit down with @jwflory.

Next attempt is going to be using the Gmail +alias-ing (such as ritlug+campusgroupslist@gmail.com which will resolve to ritlug@gmail.com but be able to used in filters) and with an additional `from` filter to prevent abuse, which will get forwarded to the list. This should allow forwarding to the list, but with enough extra safeties to prevent outside people from posting to the list as root if they do guess the alias, and so that email cc-ed by CampusGroups doesn't get posted to the list since the gmail is added as a cc email.

**Comment by justwheel** [comment]
> Next attempt is going to be using the Gmail +alias-ing (such as ritlug+campusgroupslist@gmail.com which will resolve to ritlug@gmail.com but be able to used in filters) and with an additional from filter to prevent abuse, which will get forwarded to the list. This should allow forwarding to the list, but with enough extra safeties to prevent outside people from posting to the list as root if they do guess the alias, and so that email cc-ed by CampusGroups doesn't get posted to the list since the gmail is added as a cc email.

After taking time to think about this, I think there are other things we should focus our time on more urgently than battling email – I suggest we formally deprecate Google Groups in favor of CampusGroups, and redirect to the pre-existing FOSSRIT mailing list for discussion with the wider community.

For our regular and consistent communication, using the RIT pre-baked solution for a mailing list is likely best. It helps keep us visible in the RIT system and ensures best experience from the RIT Clubs Office. However, since we already use the FOSSRIT mailing list for discussions (and there is large overlap between the Google Groups mailing list and the FOSSRIT mailing list), I don't think it is required for us to maintain our own mailing list any more. We can retire the mailing list and point non-RIT subscribers to the FOSSRIT mailing list. This also supports the use and visibility of the FOSSRIT mailing list between both RITlug members and other FOSS@MAGIC friends.

An alternative solution is to abandon Google Groups and instead forward emails to the FOSSRIT mailing list.

I think this offers us the best experience to avoid accumulating technical debt and better integrate into existing platforms used in our communities, like the FOSSRIT mailing list.

**Comment by ct-martin** [comment]
@jwflory I would like the chance to take a final try at this before we completely abandon the Google Group. The issue on my end is ability to access the RITlug root Gmail account. If this attempt fails then I have no issue with checking on the FOSSRIT list to make sure that people are fine with a notable increase in email coming from the list.

**Comment by justwheel** [comment]
I suggest we put this up to vote among all eboard members at the next eboard meeting.

**Comment by justwheel** [comment]
Removing this from the meeting agenda to wait for @ct-martin to join in person for discussion.

**Comment by ct-martin** [comment]
New blocker might be #75. Gmail has quirks for who you're allowed to forward email to. Needs more investigation. Same issues may affect using FOSS@RIT list.

**Comment by ct-martin** [comment]
Since I've exhausted most of my options for saving the Google Group, I've started reaching out about the FOSSRIT list.
I have confirmed that @ritjoe has admin access and will work with him to see if we can add CampusGroups to the list without requiring a verification email, since CampusGroups uses a no-reply email to send.

**Comment by justwheel** [comment]
@ct-martin For what it's worth, I'm also an owner of this mailing list and I'm also deeply familiar with the Mailman interface since I have configured other Postorious / Hyperkitty lists. We can write a header filter so CampusGroups emails pass through without manual moderator approval.

We can look at it tonight during FOSS Hours if you'd like. It would probably take 20-30 minutes to figure out.

**Comment by ct-martin** [comment]
@jwflory we should discuss how further another time; the from is "mailer@campusgroups.com" but ideally we should have something else checked too since it's used by more than just RITlug. I will need to look at the headers it uses and get back to you on this

**Comment by justwheel** [comment]
This one is blocked by FOSSRIT/tasks#58, since we're not sure where the FOSS@MAGIC infrastructure will be yet.
